internal/controller/k8s_resources.go

```diff
@@ -20,6 +20,14 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
+// Assuming ServiceAttributes has appropriate fields
+type Ports struct {
```

Comment:
You could replace this with `pb.KubernetesServiceData_ServicePort` directly. There's really no difference between those two structs.
If you want to clean this up in another PR, I'm OK with that.